### DISCUSSION TITLE
Can I use shadowsock to escape corporate proxy & firewall via a proxy transport?

### DISCUSSION THREAD

**Discussion by aaptel** [shadowsocks/shadowsocks-org#263] [Q&A]
I have a friend with a system in a corporate network with a very restrive firewall:

- no SSH
- only direct traffic allowed is to the corporate HTTP proxy server (80/443)
- Deep Packet Inspection

I would like to setup a program in between my system and the corporate proxy, that would encapsulate my browser requests to look like they are going to an allowed host under my control on the internet.

This host would deencapsulate the real HTTP request, and act as a proxy. It would encapsulate the response and relay it back to the corporate proxy, that would relay it back to my program, that would deencapsulate it for the browser consumption.

I have made a drawing of the current situation, and one of the setup I'm thinking of.

How can I use shadowsock for this?

![proxybypass](https://github.com/shadowsocks/shadowsocks-org/assets/1716247/6bae4004-8958-4c46-87a2-1466c7679a38)

**Comment by robot-dot-win** [comment]
Try [SoftEther](https://www.softether.org/4-docs/2-howto/7.Replacements_of_Legacy_VPNs/1.Penetrates_Firewall_by_SSL-VPN) plz :)

**Comment by sgimres** [comment]
You can try [wstunnel](https://github.com/erebe/wstunnel) 
They have an example of how to do that.
[Bypass a corporate proxy](https://github.com/erebe/wstunnel?tab=readme-ov-file#corporate)
